### PR TITLE
Add redux and react-redux

### PR DESCRIPTION
**Pull request by VitorGuedesMadeira**
### BookStore project (redux)
✔️ Add redux and redux toolkit
✔️ Structure using ''ducks'' pattern
✔️ write books/categories actions and reducers
✔️ configure redux store (configureStore)

download

### CODE REVIEW

File: src/redux/Books/Books.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// actions
+const ADD_BOOK = 'ADD_BOOK';
+const REMOVE_BOOK = 'REMOVE_BOOK';
```
Comment (Kingstalux) [inline src/redux/Books/Books.js]: - [ ] Good job so far :+1: Kindly use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for your action type names. As a result, it should be something like this:
 ```npm
const ADD_BOOK = bookstore-react/books/ADD_BOOK;
const REMOVE_BOOK = bookstore-react/books/REMOVE_BOOK;
```

File: src/redux/Books/Books.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+// reducer
+const booksReducer = (state = [], action) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return 'add a book';
```
Comment (Kingstalux) [inline src/redux/Books/Books.js]: - [ ] In your reducer, you should be adding a new book to your state and not simply returning a string called `add a book`. Kindly rectify this :+1:

File: src/redux/Books/Books.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    case REMOVE_BOOK:
+      return 'remove a book';
```
Comment (Kingstalux) [inline src/redux/Books/Books.js]: - [ ] Same issue here, you  should filter out the book you want to remove from your state and not simply return `remove a book`

File: src/redux/Categories/Categories.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+// actions
+const CHECKING_STATUS = 'Checking status...';
```
Comment (Kingstalux) [inline src/redux/Categories/Categories.js]: - [ ] Kindly apply the ducks pattern as indicated above :+1:

**Review by Kingstalux — CHANGES_REQUESTED** [review]
Hi @VitorGuedesMadeira, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights :clap: 
:ballot_box_with_check: Professional README file :100: 
:ballot_box_with_check: Good PR description :+1: 
:ballot_box_with_check: Linters are passing :ok_hand: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brytebee — APPROVED** [review]
Hi, @VitorGuedesMadeira👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Good implementation of the previous reviewer's suggestion.✔️
- App functions as it should.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 

##

###  **Optional Suggestions:**
- N/A

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**